src/PufferVaultMainnet.sol

```diff
@@ -46,22 +64,29 @@ contract PufferVaultMainnet is PufferVault {
      * Sum of EL assets + Vault Assets
      */
     function totalAssets() public view virtual override returns (uint256) {
+        // If we are dealing with native ETH deposit, we need to deduct callvalue from the balance
+        uint256 callValue;
+        assembly {
+            callValue := callvalue()
```

Comment:
Need more insights in what `callvalue()` is and why this implementation

Reply:
I've added a comment that explains this, callValue will only be populated in `depositETH` flow.

Reply:
thanks that answered all my initial questions. 

I'm wondering if this could break anything with defi integrations. Imagine a scenario that this minting/burning is happening in a chain of internal transactions (within one transaction), would this substraction create a difference between the amount minted/burned if callvalue()/msg.value is not intended for puffer contract?

Reply:
Update: seems to be a non-issue. just make sure we have tests for this